code/modules/vehicles/mecha/combat/justice.dm

```diff
@@ -0,0 +1,455 @@
+	. = ..()
+	RegisterSignal(src, COMSIG_MECHA_MELEE_CLICK, PROC_REF(justice_fatality)) //We do not hit those who are in crit or stun. We are finishing them.
+	transform = transform.Scale(1.04, 1.04)
+
+/obj/vehicle/sealed/mecha/justice/generate_actions()
+	. = ..()
+	initialize_passenger_action_type(/datum/action/vehicle/sealed/mecha/invisibility)
+	initialize_passenger_action_type(/datum/action/vehicle/sealed/mecha/charge_attack)
+
+/obj/vehicle/sealed/mecha/justice/update_icon_state()
+	. = ..()
+	if(LAZYLEN(occupants))
+		icon_state = weapons_safety ? "[base_icon_state]" : "[base_icon_state]-angry"
+	if(!has_gravity())
+		icon_state = "[icon_state]-fly"
+
+/obj/vehicle/sealed/mecha/justice/process(seconds_per_tick)
+	. = ..()
+	for(var/mob/living/occupant in occupants)
+		stealth_action = LAZYACCESSASSOC(occupant_actions, occupant, /datum/action/vehicle/sealed/mecha/invisibility)
+		charge_action = LAZYACCESSASSOC(occupant_actions, occupant, /datum/action/vehicle/sealed/mecha/charge_attack)
+		if(alpha == 255)
+			stealth_action.on = FALSE
+		else if(alpha == 0)
+			stealth_action.on = TRUE
+	update_appearance(UPDATE_ICON_STATE)
+
+/obj/vehicle/sealed/mecha/justice/set_safety(mob/user)
+	weapons_safety = !weapons_safety
+
+	if(weapons_safety)
+		movedelay = MOVEDELAY_SAFTY
+	else
+		movedelay = MOVEDELAY_ANGRY
+	playsound(src, 'sound/mecha/mech_blade_safty.ogg', 75, FALSE) //everyone need to hear this sound
+	balloon_alert(user, "justice [weapons_safety ? "calm and focused" : "is ready for battle"]")
+	SEND_SIGNAL(src, COMSIG_MECH_SAFETIES_TOGGLE, user, weapons_safety)
+	set_mouse_pointer()
+
+	update_appearance(UPDATE_ICON_STATE)
+
+/obj/vehicle/sealed/mecha/justice/Move(newloc, dir)
+	if(stealth_action.start_attack)
+		return
+	. = ..()
+	update_appearance(UPDATE_ICON_STATE)
+
+///Says 1 of 3 epic phrases before attacking and make a finishing blow to targets in stun or crit after 1 SECOND.
+/obj/vehicle/sealed/mecha/justice/proc/justice_fatality(datum/source, mob/living/pilot, atom/target, on_cooldown, is_adjacent)
+	SIGNAL_HANDLER
+
+	if(!isliving(target))
+		return
+	var/mob/living/live_or_dead = target
+	if(live_or_dead.stat < UNCONSCIOUS && live_or_dead.getStaminaLoss() < 100)
+		return FALSE
+	if(charge_action?.on)
+		return FALSE
+	if(stealth_action?.on)
+		return FALSE
+	say(pick("Take my Justice-Slash!", "A falling leaf...", "Justice is quite a lonely path"), forced = "Justice Mech")
+	playsound(src, 'sound/mecha/mech_stealth_pre_attack.ogg', 75, FALSE)
+	addtimer(CALLBACK(src, PROC_REF(finish_him), pilot, live_or_dead), 1 SECONDS)
+	return TRUE
+
+/**
+ * ## finish_him
+ *
+ * Target's head is cut off (if it has one),
+ * Otherwise it deals 100 damage.
+ * Attack from invisability and charged attack have higher priority.
+ * Arguments:
+ * * finisher - Mech pilot who makes an attack.
+ * * him - Target at which the mech makes an attack.
+ */
+/obj/vehicle/sealed/mecha/justice/proc/finish_him(mob/finisher, mob/living/him)
```

Comment:
This needs some sanity checks for:
- What if the target became alive during that 1 second?
- What if they teleported to the other side of the station during that 1 second?
- What if they were deleted during that 1 second?